pkg/chunked/storage_linux_test.go

```diff
@@ -0,0 +1,171 @@
+package chunked
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Mock for ImageSourceSeekable
+type mockImageSource struct {
+	streams chan io.ReadCloser
+	errors  chan error
+}
+
+func (m *mockImageSource) GetBlobAt(chunks []ImageSourceChunk) (chan io.ReadCloser, chan error, error) {
+	return m.streams, m.errors, nil
+}
+
+type mockReadCloser struct {
+	reader io.Reader
```

Comment:
(Absolutely non-blocking:

```go
type mockReadCloser struct {
   io.Reader
   closed bool
}
```

would allow removing the manually-written `mockReadCloser.Read` forwarder. This works just fine as is.